sprocs/ip_to_mode.sql

```diff
@@ -15,5 +18,17 @@ BEGIN
     ELSE
         mode := 'Public';
     END IF;
+
+    -- Do we have an active CBTF reservation, if so force mode to 'Exam'
+    PERFORM *
+    FROM reservations AS r
+    WHERE
+        r.user_id = ip_to_mode.authn_user_id
+        AND r.delete_date IS NULL
+        AND date BETWEEN r.access_start AND r.access_end;
```

Comment:
We don't have any good indexes on `reservations` for this sort of query, as far I can tell.

Reply:
I think an index on `(user_id, access_start)` for `reservations` would be sufficient to make this query fast. Possibly with a `WHERE delete_date IS NULL` condition.

Reply:
Also with `access_start IS NOT NULL`, I guess.

Reply:
Added the `access_start IS NOT NULL`

Did a query with the student with the biggest possible list of reservations (116 non-deleted) and the timing seems fine with the existing indexes (about 908k rows):

```
prairielearn=# EXPLAIN ANALYZE                                                                                          
SELECT * FROM reservations AS r
WHERE
    r.user_id = 34103
    AND r.delete_date IS NULL
    AND r.checked_in IS NOT NULL
    AND r.access_start IS NOT NULL
    AND '2019-09-21 02:02:02-00' BETWEEN r.access_start AND r.access_end;
                                                                                                     QUERY PLAN                                  
                                                                    
-------------------------------------------------------------------------------------------------------------------------------------------------
--------------------------------------------------------------------
 Index Scan using reservations_user_id_exam_id_delete_date_constraint on reservations r  (cost=0.42..76.46 rows=1 width=289) (actual time=0.248..
0.253 rows=1 loops=1)
   Index Cond: (user_id = 34103)
   Filter: ((checked_in IS NOT NULL) AND (access_start IS NOT NULL) AND ('2019-09-21 02:02:02+00'::timestamp with time zone >= access_start) AND 
('2019-09-21 02:02:02+00'::timestamp with time zone <= access_end))
   Rows Removed by Filter: 115
 Planning Time: 0.274 ms
 Execution Time: 0.295 ms
(6 rows)
```
Index-wise I don't think we need to change anything.